torchgeo/datasets/utils.py

```diff
@@ -737,3 +740,25 @@ def percentile_normalization(
         (img - lower_percentile) / (upper_percentile - lower_percentile + 1e-5), 0, 1
     )
     return img_normalized
+
+
+def check_instance_type(paths: Path | Iterable[Path]) -> bool:
```

Comment:
I don't love this function:

* The name is way too vague
* The docstring doesn't tell me when it's True and when it's False
* Do we really need this complex of a check? 

It seems like we're really just checking if `paths` is a `Path` (True) or `Iterable[Path]` (False). Can't we just do:
```python
return not isinstance(paths, Iterable)
```
? In that case, we wouldn't even need a function. Basically, if we can figure out a reasonable way to remove this function, I would be happier.

Reply:
@adamjstewart Unfortunately ``isinstance(paths, Iterable)`` will also return ``True`` for ``bytes and str``. will have to find another way to eliminate this function.